### PR TITLE
credential_pop typ with status-attestation-request+cwt + huge editorials

### PR DESCRIPTION
**Pull request by SaraConsoliACN**
this commit may resolve issue #20.

### CODE REVIEW

File: draft-demarco-oauth-status-attestations.md [hunk 1]
```diff
@@ -95,7 +96,7 @@ The Status Attestation is presented along with its digital credential, to prove
 
 This specification uses the terms "End-User", "Entity" as defined by
 OpenID Connect Core [@OpenID.Core], the term "JSON Web Token (JWT)"
-defined by JSON Web Token (JWT) {{RFC7519}}.
+defined by JSON Web Token (JWT) {{RFC7519}}, the term "CBOR Web Token (CWT)" defined by CBOR Web Token (CWT) {{RFC8392}}.
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:99]:
```suggestion
defined by JSON Web Token (JWT) {{RFC7519}}, the term "CBOR Web Token (CWT)" defined in {{RFC8392}}.
```

File: draft-demarco-oauth-status-attestations.md [hunk 2]
```diff
@@ -327,9 +328,9 @@ When the JWT format is used, the JWT MUST contain the parameters defined in the
 
 | JOSE Header | Description | Reference |
 | --- | --- | --- |
-| **typ** | It MUST be set to `status-attestation-request+jwt` | {{RFC7516}} Section 4.1.1 |
+| **typ** | It MUST be set to `status-attestation-request+jwt` or `status-attestation-request+cwt`. | {{RFC7516}} Section 4.1.1 |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:331]: we should say here in which cases, when JWT is must be set to ... when CWT ...

File: draft-demarco-oauth-status-attestations.md [hunk 3]
```diff
@@ -373,7 +374,7 @@ The Status Attestation MUST contain the following claims when the JWT format is
 | JOSE Header | Description | Reference |
 | --- | --- | --- |
 | **alg** | A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. It MUST NOT be set to `none` or to a symmetric algorithm (MAC) identifier. | {{RFC7515}}, {{RFC7517}} |
-| **typ** | It MUST be set to `status-attestation+jwt`. | {{RFC7515}}, {{RFC7517}} and this specification |
+| **typ** | It MUST be set to `status-attestation+jwt` or `status-attestation-request+cwt`. | {{RFC7515}}, {{RFC7517}} and this specification |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:377]: as mentioned before

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: draft-demarco-oauth-status-attestations.md (hunk 2 above)
Comment (peppelinux) [reply draft-demarco-oauth-status-attestations.md:331]: if this is a JOSE header it cannot be CWT

this is related to JOSE

File: draft-demarco-oauth-status-attestations.md [hunk 4]
```diff
@@ -327,9 +328,9 @@ When the JWT format is used, the JWT MUST contain the parameters defined in the
 
 | JOSE Header | Description | Reference |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:329]:
```suggestion
| Header Parameter Name | Description | Reference |
```

File: draft-demarco-oauth-status-attestations.md [hunk 5]
```diff
@@ -327,9 +328,9 @@ When the JWT format is used, the JWT MUST contain the parameters defined in the
...
 | --- | --- | --- |
-| **typ** | It MUST be set to `status-assertion-request+jwt` | {{RFC7516}} Section 4.1.1 |
+| **typ** | It MUST be set to `status-attestation-request+jwt` or `status-attestation-request+cwt`. | {{RFC7516}} Section 4.1.1 |
 | **alg** | A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. It MUST NOT be set to `none` or any symmetric algorithm (MAC) identifier. | {{RFC7516}} Section 4.1.1 |
-| **kid** | Unique identifier of the JWK used for the signature of the Status Attestation Request, it MUST match the one contained in the Credential `cnf.jwk`. | {{RFC7515}} |
+| **kid** | Unique identifier of the JWK used for the signature of the Status Attestation Request. It MUST match the one contained in the Credential `cnf.jwk`. | {{RFC7515}} |
 
 | JOSE Payload | Description | Reference |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:335]:
```suggestion
| Payload Parameter Name | Description | Reference |
```

File: draft-demarco-oauth-status-attestations.md [hunk 6]
```diff
@@ -373,7 +374,7 @@ The Status Assertion MUST contain the following claims when the JWT format is us
 | JOSE Header | Description | Reference |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:374]:
```suggestion
| Header Parameter Name | Description | Reference |
```

File: draft-demarco-oauth-status-attestations.md [hunk 7]
```diff
@@ -373,7 +374,7 @@ The Status Assertion MUST contain the following claims when the JWT format is us
...
 | --- | --- | --- |
 | **alg** | A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. It MUST NOT be set to `none` or to a symmetric algorithm (MAC) identifier. | {{RFC7515}}, {{RFC7517}} |
-| **typ** | It MUST be set to `status-assertion+jwt`. | {{RFC7515}}, {{RFC7517}} and this specification |
+| **typ** | It MUST be set to `status-attestation+jwt` or `status-attestation-request+cwt`. | {{RFC7515}}, {{RFC7517}} and this specification |
 | **kid** | Unique identifier of the Issuer JWK. | {{RFC7515}} |
 
 | JOSE Payload | Description | Reference |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:380]:
```suggestion
| Payload Parameter Name | Description | Reference |
```

File: draft-demarco-oauth-status-attestations.md [hunk 8]
```diff
@@ -342,25 +342,25 @@ encoding, for better readability:
 ~~~
 
 As exemplified for the JWT format, below is a non-normative example of a Credential proof of possession in CWT format. The CWT headers and payload are presented without applying signature and encoding for improved readability:
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:345]: please mention that the prosed non normative example is provided in an unencoded diagnostic representation ...

File: draft-demarco-oauth-status-attestations.md [hunk 9]
```diff
@@ -429,7 +429,7 @@ The Status Assertion MUST contain the following claims when the JWT format is us
 | **exp** | UNIX Timestamp with the expiry time of the Status Assertion. | {{RFC9126}}, {{RFC7519}} |
 | **credential_hash** | Hash value of the Digital Credential the Status Assertion is bound to. | this specification |
 | **credential_hash_alg** | The Algorithm used of hashing the Digital Credential to which the Status Assertion is bound. The value SHOULD be set to `sha-256`. | this specification |
-| **cnf** | JSON object containing the cryptographic key binding. The `cnf.jwk` value MUST match with the one provided within the related Digital Credential. | {{RFC7800}} Section 3.1 |
+| **cnf** | JSON object containing the cryptographic key binding. It MUST match with the one provided within the related Digital Credential. | {{RFC7800}} Section 3.1 |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:433]: it is not the cnf itself bu the sub-member contained in it, such as jwk for JWT and Cose Key for CWT, while there may be other parameters like `did` for instance

this must be included in this description

File: draft-demarco-oauth-status-attestations.md [hunk 10]
```diff
@@ -440,22 +464,45 @@ If the Digital Credential is valid, the Credential Issuer creates a new Status A
...
 | --- | --- | --- |
 | **iss** | It MUST be set to the identifier of the Issuer. | {{RFC9126}}, {{RFC7519}} |
 | **iat** | UNIX Timestamp with the time of the Status Assertion issuance. | {{RFC9126}}, {{RFC7519}} |
 | **exp** | UNIX Timestamp with the expiry time of the Status Assertion. | {{RFC9126}}, {{RFC7519}} |
 | **credential_hash** | Hash value of the Digital Credential the Status Assertion is bound to. | this specification |
 | **credential_hash_alg** | The Algorithm used of hashing the Digital Credential to which the Status Assertion is bound. The value SHOULD be set to `sha-256`. | this specification |
-| **cnf** | JSON object containing the cryptographic key binding. The `cnf.jwk` value MUST match with the one provided within the related Digital Credential. | {{RFC7800}} Section 3.1 |
+| **cnf** | JSON object containing the cryptographic key binding. The sub-member contained within `cnf` member, such as `jwk` for JWT and `Cose_Key` for CWT, MUST match with the one provided within the related Digital Credential. Additionally, there may be other parameters present, such as `did`, for instance. | {{RFC7800}} Section 3.1 |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:505]:
```suggestion
| **cnf** | JSON object containing the cryptographic key binding. The sub-member contained within `cnf` member, such as `jwk` for JWT and `Cose_Key` for CWT, MUST match with the one provided within the related Digital Credential. Other confirmation methods can be utilized when the referenced Digital Credential supports them, in accordance with the relevant standards. | {{RFC7800}} Section 3.1 |
```

File: draft-demarco-oauth-status-attestations.md (hunk 8 above)
Comment (SaraConsoliACN) [reply draft-demarco-oauth-status-attestations.md]: The commit that aims to resolve this is: https://github.com/peppelinux/draft-demarco-oauth-status-attestations/pull/37/commits/5980f052b048e7d7cc6aad338e7f6386646e00e5

File: draft-demarco-oauth-status-attestations.md (hunk 9 above)
Comment (SaraConsoliACN) [reply draft-demarco-oauth-status-attestations.md]: The commit that aims to resolve this is: https://github.com/peppelinux/draft-demarco-oauth-status-attestations/pull/37/commits/5980f052b048e7d7cc6aad338e7f6386646e00e5

File: draft-demarco-oauth-status-attestations.md [hunk 11]
```diff
@@ -440,22 +464,45 @@ If the Digital Credential is valid, the Credential Issuer creates a new Status A
...
 The Status Assertion MUST contain the following claims when the JWT format is used.
 
-| JOSE Header | Description | Reference |
+| Header Parameter Name | Description | Reference |
 | --- | --- | --- |
 | **alg** | A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. It MUST NOT be set to `none` or to a symmetric algorithm (MAC) identifier. | {{RFC7515}}, {{RFC7517}} |
-| **typ** | It MUST be set to `status-assertion+jwt`. | {{RFC7515}}, {{RFC7517}} and this specification |
-| **kid** | Unique identifier of the Issuer JWK. | {{RFC7515}} |
+| **typ** | It MUST be set to `status-attestation+jwt` when JWT format is used. It MUST be set to `status-attestation+cwt` when CWT format is used. | {{RFC7515}}, {{RFC7517}} and this specification |
+| **kid** | Unique identifier of the Issuer JWK or CWT. | {{RFC7515}} |
```
[the remark this answers is not in the document]
Comment (peppelinux) [reply draft-demarco-oauth-status-attestations.md:496]: JWT != JWK therefore != CWT

File: draft-demarco-oauth-status-attestations.md [hunk 12]
```diff
@@ -369,8 +372,25 @@ MUST demonstrate the proof of possession of the Digital Credential to
...
-referenced in the Digital Credential.
+object (JWT or CWT) signed with the private key that its related public key is referenced in the Digital Credential.
+
+When the JWT or CWT format are used, the JWT/CWT MUST contain the parameters defined in the following table.
+
+| Header | Description | Reference |
+| --- | --- | --- |
+| **typ** | It MUST be set to `status-attestation+jwt` when JWT format is used. It MUST be set to `status-attestation+cwt` when CWT format is used. | {{RFC7516}} Section 4.1.1 |
+| **alg** | A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. It MUST NOT be set to `none` or any symmetric algorithm (MAC) identifier. | {{RFC7516}} Section 4.1.1 |
+| **kid** | Unique identifier of the JWK or CWT used for the signature of the Status Attestation Request, it MUST match the one contained in the Credential. | {{RFC7515}} |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:383]:
```suggestion
| **kid** | Unique identifier of the JWK or "Cose Key" used for the signature of the Status Attestation Request, it MUST match the one contained in the Credential. | {{RFC7515}} |
```

JWT != JWK and therefore != CWT

please check Cose key if good.

File: draft-demarco-oauth-status-attestations.md [hunk 13]
```diff
@@ -369,8 +372,25 @@ MUST demonstrate the proof of possession of the Digital Credential to
...
+
+| Payload | Description | Reference |
+| --- | --- | --- |
+| **iss** | Wallet identifier. | {{RFC9126}}, {{RFC7519}} |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:387]:
```suggestion
| **iss** | Wallet identifier. | {{RFC9126}}, {{RFC7519}} |
```

concerns here, since a wallet identifier can be a vector of traceability. We need further explaination about the evidence that this identifier is not unique and absolute, preventing the traceability of the wallet.

File: draft-demarco-oauth-status-attestations.md [hunk 14]
```diff
@@ -369,8 +372,25 @@ MUST demonstrate the proof of possession of the Digital Credential to
 the Credential Issuer.
 
 The proof of possession is made by enclosing the Digital Credential in an
-object (JWT) signed with the private key that its related public key is
-referenced in the Digital Credential.
+object (JWT or CWT) signed with the private key that its related public key is referenced in the Digital Credential.
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:375]: I would rephrase saying using the same confirmation method referenced ... 

therefore using the elements defined in RFC7800

File: draft-demarco-oauth-status-attestations.md [hunk 15]
```diff
@@ -440,22 +464,45 @@ If the Digital Credential is valid, the Credential Issuer creates a new Status A
 }
 ~~~
 
+As exemplified for the JWT format, below is a non-normative example of a Credential proof of possession in CWT format provided in an unencoded diagnostic representation. The CWT headers and payload are presented without applying signature and encoding for improved readability:
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:467]:
```suggestion
As exemplified for the JWT format, below is a non-normative example of a Status Assertion Request object in CWT format provided in CBOR diagnostic notation format, as defined in {{rfc8152}} where the CWT headers and payload are represented without applying signature and encoding for better readability:
```

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: draft-demarco-oauth-status-attestations.md (hunk 14 above)
Comment (SaraConsoliACN) [reply draft-demarco-oauth-status-attestations.md]: The commit https://github.com/peppelinux/draft-demarco-oauth-status-attestations/pull/37/commits/5f23e8e156d732e1d2fbad89b81b62b7952f69fd aims to resolve this comment.

File: draft-demarco-oauth-status-attestations.md [hunk 16]
```diff
@@ -384,7 +385,8 @@ When the JWT or CWT format are used, the JWT/CWT MUST contain the parameters def
 
 | Payload | Description | Reference |
 | --- | --- | --- |
-| **iss** | Wallet identifier. | {{RFC9126}}, {{RFC7519}} |
+| **iss** | Wallet identifier. The Wallet identifier is out of the scope of this specs considering that it may be just a
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:388]: the wallet identifier is not a framework

File: draft-demarco-oauth-status-attestations.md [hunk 17]
```diff
@@ -502,7 +504,7 @@ The Status Assertion MUST contain the following claims when the JWT format is us
...
 | **credential_hash** | Hash value of the Digital Credential the Status Assertion is bound to. | this specification |
 | **credential_hash_alg** | The Algorithm used of hashing the Digital Credential to which the Status Assertion is bound. The value SHOULD be set to `sha-256`. | this specification |
-| **cnf** | JSON object containing the cryptographic key binding. The sub-member contained within `cnf` member, such as `jwk` for JWT and `Cose_Key` for CWT, MUST match with the one provided within the related Digital Credential. Other confirmation methods can be utilized when the referenced Digital Credential supports them, in accordance with the relevant standards. | {{RFC7800}} Section 3.1 |
+| **cnf** | JSON object containing the cryptographic key binding. The sub-member contained within `cnf` member, such as `jwk` for JWT and `Cose_Key` for CWT, MUST match with the one provided within the related Digital Credential. Other confirmation methods can be utilized when the referenced Digital Credential supports them, in accordance with the relevant standards. | {{RFC7800}} Section 3.1, {{RFC8747}} Section 3.1 |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:507]: here, we must use the term "confirmation methods" to not use only cryptographic key bindings

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: draft-demarco-oauth-status-attestations.md (hunk 16 above)
Comment (fmarino-ipzs) [reply draft-demarco-oauth-status-attestations.md:388]: The last sentence ("considering that....") is not clear to me. Is there a reason why we decided to emphasize that the Wallet identifier is out of the scope of this spec? I suggest removing this sentence., if you agree.
@OR13 @marinaado

File: draft-demarco-oauth-status-attestations.md (hunk 17 above)
Comment (SaraConsoliACN) [reply draft-demarco-oauth-status-attestations.md]: Resolved with the commit https://github.com/peppelinux/draft-demarco-oauth-status-attestations/pull/37/commits/ebd064bad946d0a1315417307c366642a6f625f7

File: draft-demarco-oauth-status-attestations.md (hunk 16 above)
Comment (peppelinux) [reply draft-demarco-oauth-status-attestations.md:388]: @fmarino-ipzs the way a wallet can be identified or the value to be supposed to be used for identify a wallet is definitively out of the scope of this spec

File: draft-demarco-oauth-status-attestations.md [hunk 18]
```diff
@@ -368,9 +372,27 @@ Status Assertion,
 MUST demonstrate the proof of possession of the Digital Credential to
 the Credential Issuer.
 
-The proof of possession is made by enclosing the Digital Credential in an
-object (JWT) signed with the private key that its related public key is
-referenced in the Digital Credential.
+The proof of possession is made by enclosing the Digital Credential
+in an object (JWT or CWT) using the same confirmation method referenced in RFC7800.
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:376]: the digital credential is not enclosed, an hash of it is provided in the status assertion request and the same hash in the status assertion response

File: draft-demarco-oauth-status-attestations.md [hunk 19]
```diff
@@ -368,9 +372,27 @@ Status Assertion,
...
+| **iss** | Wallet identifier. The Wallet identifier is out of the scope of this specs considering that it may be just a
+framework that renders the Wallet  identifier itself ephemeral. | {{RFC9126}}, {{RFC7519}} |
+| **aud** | It MUST be set with the Credential Issuer Status Attestation endpoint URL as value that identify the intended audience | {{RFC9126}}, {{RFC7519}} |
+| **exp** | UNIX Timestamp with the expiration time of the JWT. It MUST be superior to the value set for `iat`  . | {{RFC9126}}, {{RFC7519}}, {{RFC7515}} |
+| **iat** | UNIX Timestamp with the time of JWT/CWT issuance. | {{RFC9126}}, {{RFC7519}} |
+| **jti** | Unique identifier for the JWT.  | {{RFC7519}} Section 4.1.7 |
+| **credential_hash** | Hash value of the Digital Credential the Status Attestation is bound to. | this specification |
+| **credential_hash_alg** |  The Algorithm used of hashing the Digital Credential to which the Status Attestation is bound. The value SHOULD be set to `sha-256`. | this specification |
 
 Below is a non-normative example of a Credential proof of possession with
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:397]:
```suggestion
Below is a non-normative example of a Status Assertion Request with
```

**Review by peppelinux — APPROVED** [review]

**Comment by peppelinux** [comment]
This PR ios ready to be merged, a further PR will be create to address @marinaado 's comment  https://github.com/peppelinux/draft-demarco-oauth-status-attestations/pull/45#issuecomment-2117623296 and the table related to the errors where the placeholder is here defined: https://github.com/peppelinux/draft-demarco-oauth-status-attestations/pull/37/files#diff-6018989306b8996e79d471eb123fc3b59796d1635384b6faac8cdb36cd5f49ccR472

regarding the first comment, since it is required that the status assertion issuer always respond with an application/json containing the status_assertion_errors, I'd try to use a general approach without defining specialized error section for both the requests and the responses

File: draft-demarco-oauth-status-attestations.md [hunk 20]
```diff
@@ -57,21 +59,19 @@ without requiring to query any third-party entities.
 
 # Introduction
 
-Status Assertions ensure the integrity and trustworthiness of digital
+Status Assertions ensure the non-revocation of digital
```
Comment (OR13) [inline draft-demarco-oauth-status-attestations.md:62]: is there interest in suspension, revocation or other dynamic state? see https://w3c.github.io/vc-bitstring-status-list/#example-example-statuslistcredential-using-more-complex-entries

Some parties have use cases that include dynamic state that is an enumeration instead of a boolean.

**Review by OR13 — APPROVED** [review]

File: draft-demarco-oauth-status-attestations.md (hunk 20 above)
Comment (peppelinux) [reply draft-demarco-oauth-status-attestations.md:62]: thank you, moved to an issue

Comment (peppelinux) [reply draft-demarco-oauth-status-attestations.md:62]: https://github.com/peppelinux/draft-demarco-oauth-status-attestations/issues/49

File: draft-demarco-oauth-status-attestations.md [hunk 21]
```diff
@@ -309,132 +306,181 @@ related to a specific Credential issued by the same Credential Issuer.
...
+Status Assertions and or the Status Assertion Errors
+related to the request made by the
+Wallet Instance. In the non-normative example below is
+represented an HTTP Response with the
+`status_assertion_responses` JSON member:
 
 ~~~
-  HTTP/1.1 400 Bad Request
-  Content-Type: application/json;charset=UTF-8
+HTTP/1.1 201 Created
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:425]:
```suggestion
HTTP/1.1 200 Created
```

File: draft-demarco-oauth-status-attestations.md [hunk 22]
```diff
@@ -309,132 +306,181 @@ related to a specific Credential issued by the same Credential Issuer.
...
 
-Below is a non-normative example of a Credential proof of possession with
-the JWT headers and payload are represented without applying signature and
-encoding, for better readability:
+For each entry in the `status_assertion_responses` array, the following requirements are met:
+- Each element in the array MUST match the corresponding element in the request array at
+the same position index to which it is related, eg: "[requestAboutA, requestAboutB]" produces "[responseAboutA, responseErrorAboutB]".
+- Each element MUST contain the error or the status of the assertion using the `typ` member.
+set to "status-assertion+{jwt,cwt}" or "status-assertion-error+{jwt,cwt}", depending by the object type.
+- The corresponding entry in the response MUST be of the same type as requested. For example,
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:444]:
```suggestion
- The corresponding entry in the response MUST be of the same data format as requested. For example,
```

File: draft-demarco-oauth-status-attestations.md [hunk 23]
```diff
@@ -309,132 +306,181 @@ related to a specific Credential issued by the same Credential Issuer.
...
+
+| Header | Description | Reference |
+| --- | --- | --- |
+| **typ** | It MUST be set to `status-attestation+jwt` when JWT format is used. It MUST be set to `status-attestation+cwt` when CWT format is used. | {{RFC7516}} Section 4.1.1 |
+| **alg** | A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. It MUST NOT be set to `none` or any symmetric algorithm (MAC) identifier. | {{RFC7516}} Section 4.1.1 |
+| **kid** | Unique identifier of the `JWK or` `Cose_Key` used for the signature of the Status Attestation Request, it MUST match the one contained in the Credential. | {{RFC7515}} |
+
+| Payload | Description | Reference |
+| --- | --- | --- |
+| **iss** | Wallet identifier. The Wallet identifier value is supposed to be used for identifying a wallet; therefore, it is out of scope for this specs, considering that it may simply be an ephemeral value. | {{RFC9126}}, {{RFC7519}} |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:326]:
```suggestion
| **iss** | Status Assertion Request Issuer identifier. The value is supposed to be used for identifying the Wallet that has issued the request. It is out of scope for this document defining how this value must be set. | {{RFC9126}}, {{RFC7519}} |
```

File: draft-demarco-oauth-status-attestations.md [hunk 24]
```diff
@@ -309,132 +306,181 @@ related to a specific Credential issued by the same Credential Issuer.
...
+| **aud** | It MUST be set with the Credential Issuer Status Attestation endpoint URL as value that identify the intended audience | {{RFC9126}}, {{RFC7519}} |
+| **exp** | UNIX Timestamp with the expiration time of the JWT. It MUST be superior to the value set for `iat`  . | {{RFC9126}}, {{RFC7519}}, {{RFC7515}} |
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:328]:
```suggestion
| **aud** | It MUST be set with the Credential Issuer Status Attestation endpoint URL as value that identify the intended audience. | {{RFC9126}}, {{RFC7519}} |
| **exp** | UNIX Timestamp with the expiration time of the JWT. It MUST be superior to the value set for `iat` . | {{RFC9126}}, {{RFC7519}}, {{RFC7515}} |
```

File: draft-demarco-oauth-status-attestations.md [hunk 25]
```diff
@@ -309,132 +306,181 @@ related to a specific Credential issued by the same Credential Issuer.
...
+}
+.
+{
+    "iss": "0b434530-e151-4c40-98b7-74c75a5ef760",
+    "aud": "https://issuer.example.org/status-assertion-endpoint",
+    "iat": 1698744039,
+    "exp": 1698830439,
+    "jti": "6f204f7e-e453-4dfd-814e-9d155319408c",
+    "credential_hash": $Issuer-Signed-JWT-Hash
+    "credential_hash_alg": "sha-256",
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:352]:
```suggestion
    "credential_hash_alg": "sha-256"
```

**Review by fmarino-ipzs — APPROVED** [review]
